torch/csrc/jit/profiling_graph_executor_impl.cpp

```diff
@@ -11,29 +18,65 @@ bool& getProfilingMode() {
   return profiling_mode;
 }
 
-static std::shared_ptr<Graph> prepareGraph(const std::shared_ptr<Graph>& graph) {
+std::shared_ptr<Graph> ProfilingGraphExecutorImpl::prepareGraph(
+    const std::shared_ptr<Graph>& graph,
+    Stack& stack) {
   auto g = graph->copy();
+  ArgumentSpec spec =
```

Comment:
With the profiling graph executor, we no longer need ArgumentSpec, so we should not carry it over here. I'd like to delete it, as its complexity is becoming a maintenance burden.

Reply:
This is fine for WIP code though, since I know this relates to the PropagateRequiresGrad pass. However, we should edit propagates requires grad so that it works with ProfiledTensorType as well and can read the grad information from there.

Reply:
Yes, I had to use it since it fills in `requires_grad` for inputs. I'll try to detangle `requires_grad` propagation in a separate PR.